draft-ietf-webtrans-http3.md

```diff
@@ -568,6 +543,298 @@ A TLS exporter API might permit the context field to be omitted. In this case,
+connection if the client opens sessions exceeding this limit, as the client and
+the server do not have a consistent view of how many sessions are open due to
+the asynchronous nature of the protocol; instead, it MUST reset all of the
+CONNECT streams it is not willing to process with the `HTTP_REQUEST_REJECTED`
+status defined in [HTTP3].
+
+Just like other HTTP requests, WebTransport sessions, and data sent on those
+sessions, are counted against flow control limits.  This document does not
+introduce additional mechanisms for endpoints to limit the relative amount of
+flow control credit consumed by different WebTransport sessions, however
+servers that wish to limit the rate of incoming requests on any particular
+session have alternative mechanisms:
+
+* The `HTTP_REQUEST_REJECTED` error code defined in [HTTP3] indicates to the
+  receiving HTTP/3 stack that the request was not processed in any way.
+* HTTP status code 429 indicates that the request was rejected due to rate
+  limiting {{!RFC6585}}.  Unlike the previous method, this signal is directly
+  propagated to the application.
+
+## Limiting the Number of Streams Within a Session {#flow-control-limit-streams}
+
+The WT_MAX_STREAMS capsule ({{WT_MAX_STREAMS}}) establishes a limit on the
+number of streams within a WebTransport session.  Like the QUIC MAX_STREAMS
+frame ({{Section 19.11 of !RFC9000}}), this capsule has two types that provide
+separate limits for unidirectional and bidirectional streams that are initiated
+by a peer.
+
+The session-level stream limit applies in addition to the QUIC MAX_STREAMS
+frame, which provides a connection-level stream limit.  New streams can only be
+created within the session if both the stream- and the connection-level limit
+permit; see {{Section 4.6 of !RFC9000}} for details on how QUIC stream limits
+are applied.
+
+Unlike the WT_MAX_STREAMS capsule or the QUIC MAX_STREAMS frame, there is no
+simple relationship between the value in this frame and stream IDs in QUIC
+STREAM frames.  This especially applies if there are other users of streams on
+the connection.
+
+The WT_STREAMS_BLOCKED capsule ({{WT_STREAMS_BLOCKED}}) is sent to indicate that
+an endpoint was unable to create a stream due to the session-level stream
+limit.
+
+## Data Limits {#flow-control-limit-data}
+
+The WT_MAX_DATA capsule ({{WT_MAX_DATA}}) establishes a limit on the amount of
+data that can be sent within a WebTransport session.  This limit counts all data
+that is sent on streams of the corresponding type, excluding the stream header
+(see {{unidirectional-streams}} and {{bidirectional-streams}}).  The stream
+header is excluded from this limit so that this limit does not prevent the
+sending of information that is essential in linking new streams to a specific
+WebTransport session.
+
+Implementing WT_MAX_DATA requires that the QUIC stack provide the WebTransport
+implementation with information about the final size of streams; see {{Section
+4.5 of !RFC9000}}.
```

Comment:
Is it worth highlighting this as a difference?  With HTTP/2 you don't need this, but HTTP/3 means that you are creating streams all over and using final size.  We might also note that this is independent of any RESET_STREAM_AT functionality.

Reply:
When you say highlighting this as a difference, what're you thinking? Like adding additional text to describe how this works here as opposed to H2? (I'd imagine we'd add that to H2 where we refer to this capsule?)